### DISCUSSION TITLE
papis_libgen

### DISCUSSION THREAD

**Discussion by UromPLAt** [papis/papis#380] [Q&A] [answered]
Hi,
What does the papis_libgen do?
Just wondering..
Couldn't find an answer in the readme or the docs..

**Comment by manuelhaussmann** [comment]
Libgen works similar to scihub which you might have come across. (See here for some wiki information: https://en.wikipedia.org/wiki/Library_Genesis)

**Reply by UromPLAt** [reply]
Sorry, let me clarify, I know these services (libgen, scihub), I just don't understand how Papis interfaces with them.

I looked at the plugin code now and it seems the plugin can query libgen by author, title or ISBN.

There seems to be an option to download by MD5 hash but I don't understand Python well enough to say for sure..

**Reply by manuelhaussmann** [reply]
Got it. On that part I can't comment, I haven't used it myself yet.

**Comment by alexfikl** [comment, marked as answer]
From a quick skim through the [code](https://github.com/papis/papis-libgen/blob/master/papis_libgen/__init__.py), it seems to be using [pylibgen](https://pypi.org/project/pylibgen/) as a backend and provides
* an [explore plugin](https://papis.readthedocs.io/en/latest/commands.html#module-papis.commands.explore) to search for documents.
* a `Downloader` to actually download documents (should work with `papis add`).

It seems that `pylibgen` is unmaintained though and `papis_libgen` could also use some love, so not sure if they actually work (I haven't used it myself either).

Hope that helps!

**Reply by alejandrogallo** [reply]
this is pretty much it, both libraries `pylibgen` and `papis_libgen` would benefit from a maintainer
as stated in https://github.com/papis/papis#papis-projects-maintainers
it should not be a lot of work in general

it could also be that by now there are some better alternatives than pylibgen, but I have not looked into it and I do not intend to

**Comment by Archie-2021** [comment]
Any development on this? I would like to fetch books from libgen like this

**Reply by alexfikl** [reply]
I'm afraid not. There doesn't seem to be anyone interested in maintaining it :(

**Reply by alexfikl** [reply]
There seem to be some tools for libgen here: https://github.com/freereadorg/awesome-libgen
Maybe you can use those to get the documents and then add them to your Papis library (manually :(.

**Reply by alexfikl** [reply]
Also https://github.com/harrison-broadbent/libgen-api seems to be a maintained python library for libgen that we could use :grin: Just need a generous soul to look at `papis_libgen`!
